skrifa/src/color/mod.rs

```diff
@@ -218,7 +237,27 @@ pub trait ColorPainter {
     fn pop_clip(&mut self);
 
     /// Fill the current clip area with the specified gradient fill.
-    fn fill(&mut self, brush: Brush);
+    fn fill(&mut self, brush: Brush<'_>);
+
+    /// Queries from the client whether the optimized
+    /// [`fill_glyph()`](ColorPainter::fill_glyph) operation is supported.
+    fn fill_glyph_supported(&self) -> FillGlyph {
+        FillGlyph::Unimplemented
+    }
+
+    /// Combined clip and fill operation.
+    ///
+    /// Apply the clip path determined by the specified `glyph_id`,
+    /// then fill it with the specified [`brush`](Brush), applying the `_transform`
+    /// transformation matrix to the brush.
+    fn fill_glyph(
+        &mut self,
+        _glyph_id: GlyphId,
+        _transform: Transform,
```

Comment:
This should probably be called `brush_transform` just to further signify that it applies to the brush and not the clip. 

Also, optional: make the transform optional :) I suspect this will be identity most of the time.

Reply:
a) Renamed to `brush_transform` and updated arg name in documentation.
b) Transform turned into `Option<Transform>` - I wasn't sure if you prefer to have that be expressed in the test results as an empty string - for now I left testing to dump an identity matrix in that case, instead of special casing the None situation,  but agree this might be helpful for clients. Let me know if you want changes on the testing side. On the Skia side I am also making this a default matrix in the callbacks to C++.

Reply:
Your approach looks good to me. Thanks for making the change.